code/game/objects/items/stacks/golem_food/golem_status_effects.dm

```diff
@@ -95,6 +108,20 @@
 	QDEL_LIST(active_overlays)
 	return ..()
 
+// Golem's arms are central to their status effects so the golem effects have procs for setting their arms
+/datum/status_effect/golem/proc/set_arm_effects()
+	SHOULD_CALL_PARENT(TRUE)
+
+	for(var/obj/item/bodypart/arm/arm in human_owner.bodyparts)
+		if(arm.limb_id != SPECIES_GOLEM)
+			continue
+		LAZYADD(modified_arms, arm)
```

Comment:
Should probably also have the memory management qdel signal handling from the diamond ore (perhaps alos others?) here instead.  
If the arm is somehow deleted it will still otherwise exist in this list and cause a hard deletion, unless you track it being deleted and remove it from the list at this point.